static/app/views/settings/project/sampling/modal/index.tsx

```diff
@@ -1,86 +1,234 @@
+import useApi from 'sentry/utils/useApi';
+import {useLocation} from 'sentry/utils/useLocation';
+import {useParams} from 'sentry/utils/useParams';
+import {useRoutes} from 'sentry/utils/useRoutes';
+import EmptyMessage from 'sentry/views/settings/components/emptyMessage';
+import TextBlock from 'sentry/views/settings/components/text/textBlock';
 
-import {SAMPLING_DOC_LINK} from '../utils';
+import {getInnerNameLabel, isCustomTagName} from '../utils';
 
-import RuleModal from './ruleModal';
+import Conditions from './conditions';
 import {
   distributedTracesConditions,
   generateConditionCategoriesOptions,
+  getErrorMessage,
   getNewCondition,
   individualTransactionsConditions,
+  isLegacyBrowser,
 } from './utils';
 
-type RuleModalProps = React.ComponentProps<typeof RuleModal>;
-
-type Props = Omit<
-  RuleModalProps,
-  | 'transactionField'
-  | 'title'
-  | 'conditionCategories'
-  | 'onSubmit'
-  | 'emptyMessage'
-  | 'onChage'
-> & {
+const conditionAlreadyAddedTooltip = t('This condition has already been added');
+
+type ConditionsProps = React.ComponentProps<typeof Conditions>['conditions'];
+
+type State = {
+  conditions: ConditionsProps;
+  errors: {
+    sampleRate?: string;
+  };
+  sampleRate: number | null;
+};
+
+type Props = ModalRenderProps & {
+  disabled: boolean;
+  onSubmitSuccess: (project: Project, successMessage: React.ReactNode) => void;
+  organization: Organization;
+  project: Project;
   rules: SamplingRules;
+  type: SamplingRuleType;
+  rule?: SamplingRule;
 };
 
-function TransactionRuleModal({rule: ruleToUpdate, rules, ...props}: Props) {
-  const theme = useTheme();
+export function SamplingRuleModal({
+  Header,
+  Body,
+  Footer,
+  closeModal,
+  project,
+  onSubmitSuccess,
+  rule,
+  rules,
+  disabled,
+  type,
+  organization,
+}: Props) {
+  const api = useApi();
+  const params = useParams();
+  const location = useLocation();
+  const routes = useRoutes();
 
-  const [tracing, setTracing] = useState(
-    ruleToUpdate ? ruleToUpdate.type === SamplingRuleType.TRACE : true
-  );
+  const [data, setData] = useState<State>(getInitialState());
+  const [isSaving, setIsSaving] = useState(false);
 
-  const [isTracingDisabled, setIsTracingDisabled] = useState(
-    !!ruleToUpdate?.condition.inner.length
+  const conditionCategories = generateConditionCategoriesOptions(
+    type === SamplingRuleType.TRACE
+      ? distributedTracesConditions
+      : individualTransactionsConditions
   );
 
-  function handleChange({
-    conditions,
-  }: Parameters<NonNullable<RuleModalProps['onChange']>>[0]) {
-    setIsTracingDisabled(!!conditions.length);
+  useEffect(() => {
+    setData(d => {
+      if (!!d.errors.sampleRate) {
+        return {...d, errors: {...d.errors, sampleRate: undefined}};
+      }
+
+      return d;
+    });
+  }, [data.sampleRate]);
+
+  function getInitialState(): State {
+    if (rule) {
+      const {condition: conditions, sampleRate} = rule as SamplingRule;
```

Comment:
```suggestion
      const {condition: conditions, sampleRate} = rule;
```

We can drop `as SamplingRule` because if it's defined, its type is `SamplingRule`